doc/workshops/plugin-development.rst

```diff
@@ -2,4 +2,917 @@
+
+
+
+=========================
+Building our first plugin
+=========================
+
+
+Widgets
+~~~~~~~
+
+Best is going backwards to build our plugin, let's start by implementing the graphical controls of our plugin in ``widgets.py``
+
+Originally, we have:
+
+.. code-block:: python
+
+   # Spyder imports
+   from spyder.api.config.decorators import on_conf_change
+   from spyder.api.translations import get_translation
+
+   from spyder.api.widgets.mixins import SpyderWidgetMixin
+
+
+   # Localization
+   _ = get_translation("spyder_pomodoro_timer.spyder")
+
+The preset imports are a guide to what we will need in our plugin. The ``on_conf_change`` decorator will allow us to propagate the changes in the configuration. ``get_translation`` help us to generate the translation strings and ``SpyderWidgetMixin`` adds to any Widget the attributes and methods needed to integrate with Spyder (icon, style, translation, actions and extra options).
+
+After some research in the Spyder source code we can find that in Spyder there are two types of predefined components for the status bar:
+
+* ``StatusBarWidget``, a class derived from ``QWidget`` and ``SpyderWidgetMixin``, which contains an icon, a label and a spinner (to show the plugin loading).
+* ``BaseTimerStatus``, a class derived from ``StatusBarWidget`` with an internal ``QTimer`` to periodically update its content.
+
+Since we want a widget that is periodically updated showing the pomodoro countdown, we will use a ``BaseTimerStatus`` instance.
+
+So, we can safely substitute
+
+.. code-block:: python
+
+   from spyder.api.widgets.mixins import SpyderWidgetMixin
+
+with
+
+.. code-block:: python
+
+   from spyder.api.widgets.status import BaseTimerStatus
+   from spyder.utils.icon_manager import ima
+
+Add an initial import:
+
+.. code-block:: python
+
+   # Third party imports
+   import qtawesome as qta
+
+And add at the end of the file:
+
+.. code-block:: python
+
+   class PomodoroTimerStatus(BaseTimerStatus):
+       """Status bar widget to display the pomodoro timer"""
+
+       ID = "pomodoro_timer_status"
+       CONF_SECTION = "pomodoro_timer"
+
+       def __init__(self, parent):
+           super().__init__(parent)
+           self.value = "25:00"
+
+       def get_tooltip(self):
+           """Override api method."""
+           return "I am the Pomodoro tooltip"
+
+       def get_icon(self):
+           return qta.icon("mdi.av-timer", color=ima.MAIN_FG_COLOR)
+
+Spyder needs ``ID`` and ``CONF_SECTION`` to be defined, the constructor calls the parent class constructor and initializes the label ``value``.
+
+We add a tooltip to verify the existence of our plugin. Since Spyder uses ``qtawesome`` (is another spyder team project which makes it easy incorporate iconic fonts into our interface) we can select our icon using ``qta-browser`` from the terminal.
```

Comment:
```suggestion
We add a tooltip to verify the existence of our plugin. Since Spyder uses ``qtawesome`` (a project which makes it easy incorporate iconic fonts into our interface) we can select our icon using ``qta-browser`` from the terminal.
```
Add link to the qtawesome repo